packages/react-reconciler/src/__tests__/ReactIncrementalPerf-test.internal.js

```diff
@@ -584,6 +584,7 @@ describe('ReactDebugFiberPerf', () => {
       const MemoFoo = React.memo(function Foo() {
         return <div />;
       });
+      MemoFoo.displayName = 'Foo';
```

Comment:
[`beginFiberMark`](https://github.com/facebook/react/blob/f3a14951ab9bccfd59ca977493b72321b24e50a5/packages/react-reconciler/src/ReactDebugFiberPerf.js#L122) will be called with the initial Fiber. However, in [`updateMemoComponent`](https://github.com/facebook/react/blob/4c78ac0b9df88edec73492f92f09d5438dd74c4d/packages/react-reconciler/src/ReactFiberBeginWork.js#L352) we will upgrade the Fiber to a `SimpleMemoComponent` at which point the Fiber is indistinguishable for `getComponentName` from a function component because with mutated the type. [`endFiberMark`](https://github.com/facebook/react/blob/f3a14951ab9bccfd59ca977493b72321b24e50a5/packages/react-reconciler/src/ReactDebugFiberPerf.js#L154) will receive a `SimpleMemoComponent` and `getComponentName` assumes it's a plain function component.

Not sure if this is considered a defect (assuming that the fiber type in `beginFiberWork` is the same we get in `endFiberWork`). Should we we add something `_debugOriginalType` field to the Fiber or rather change `react-devtools` to display `Foo` rather than `Memo(Foo)`? Mutating the `displayName` of the type when upgrading the Fiber might also work.

Any thoughts @bvaughn ?